src/mctp-netlink.c

```diff
@@ -722,8 +725,17 @@ static int parse_getlink_dump(mctp_nl *nl, struct nlmsghdr *nlh, uint32_t len)
 			continue;
 		}
 		ifname_len = strnlen(ifname, ifname_len);
+
+		ifaddr = mctp_get_rtnlmsg_attr(IFLA_ADDRESS, rta, rlen,
+					       &ifaddr_len);
+		if (!ifaddr) {
+			warnx("no ifaddr?");
```

Comment:
This is not a warning, nor should we skip setting up the link - some links may not have a physical address.

There are two possible (valid) behaviours there - that there is no `IFLA_ADDRESS` attribute, or that the length is reported as zero. We should allow both of those.

Reply:
Makes sense. Code that uses `ifaddr` should have already taken `ifaddr_len` into account (like `memcpy`), so I will remove this check.